libs/StatusGoQt/src/StatusGo/Wallet/WalletApi.cpp

```diff
@@ -34,6 +34,35 @@ DerivedAddresses getDerivedAddressesForPath(const HashedPassword &password, cons
     return resultJson.get<CallPrivateRpcResponse>().result;
 }
 
+SavedAddresses getSavedAddresses()
+{
+    json inputJson = {
+        {"jsonrpc", "2.0"},
+        {"method", "wallet_getSavedAddresses"}
+    };
+
+    auto result = Utils::statusGoCallPrivateRPC(inputJson.dump().c_str());
+    const auto resultJson = json::parse(result);
+    checkPrivateRpcCallResultAndReportError(resultJson);
+
+    const auto &data = resultJson.get<CallPrivateRpcResponse>().result;
```

Comment:
`CallPrivateRpcResponse` should not be destroyed until function exit, but this pard definitely should be improved - not only in this procedure

Reply:
For sure lifetime of the temporary would be extended in this case:
```
const auto &data = resultJson.get<CallPrivateRpcResponse>();
```
but in our case looks like taking reference to a member of `CallPrivateRpcResponse`'s object which is destroyed immediately. I would opt for double checking it because it can lead to very hard to nail down bugs...

Reply:
Ok, false alarm. This code is almost for sure ok. It's nicely explained here: https://quuxplusone.github.io/blog/2020/11/16/lifetime-extension-tidbit/